core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -2562,6 +2692,81 @@ private LocalSyncWriteModelContainer replaceOrUpsertOneFromRemote(
     return container;
   }
 
+  /**
+   * Updates a single synchronized document by its given id with the given update description.
+   *
+   * @param nsConfig  the namespace synchronization config of the namespace where the document
+   *                  lives.
+   * @param documentId the _id of the document.
+   * @param updateDescription the update description to apply locally.
+   */
+  @CheckReturnValue
+  private LocalSyncWriteModelContainer updateOneFromRemote(
+      final NamespaceSynchronizationConfig nsConfig,
+      final BsonValue documentId,
+      final UpdateDescription updateDescription,
+      final BsonDocument atVersion,
+      final long withHash
+  ) {
+    if (updateDescription.isEmpty()) {
+      // don't do anything for completely no-op updates
+      return null;
+    }
```

Comment:
Was thinking you may want to do that after the call to sanitizeUpdateDescription()

Reply:
As discussed offline, this needs to be here since an empty sanitized update description might still have version changes that make it not a no-op.